### PR TITLE
Week5 stage 1 

### PR DESCRIPTION
**Pull request by joudi6**
Hi Jalal! sry for late handing. I deleted my app and started from scratch again today with a new algorithm and different component structuring. thank you for your patience.

### CODE REVIEW

File: homework-submission/photo-wall-app/src/App.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import React, { Component } from 'react';
+import './App.css';
+import LoginForm from './components/login-form'
+import Post from './components/post'
+
+class App extends Component {
+  state = {
+    loading: false,
```
Comment (jalal75) [inline homework-submission/photo-wall-app/src/App.js]: this variable is not used.

File: homework-submission/photo-wall-app/src/components/login-form.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    user: ''
+  }
+  handleInputChange = (event) => {
+    this.setState({ user: event.target.value })
+    console.log(this.state.user);
+
+  }
+  handleLogin = (event) => {
+    event.preventDefault();
+    (this.state.user.length > 0 && this.state.user !== 'User Name') ? this.props.onLogin(this.state.user) : alert('type your User Name')
```
Comment (jalal75) [inline homework-submission/photo-wall-app/src/components/login-form.js]: this.state.user !== 'User Name'
Why.....?

File: homework-submission/photo-wall-app/src/components/post.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import AddForm from './add-from'
+import Service from './service'
+import Header from './header'
+
+class Post extends Component {
+  state = {
+    loading: false,
+    author: '',
+    posts: [],
+    form: 'list'
```
Comment (jalal75) [inline homework-submission/photo-wall-app/src/components/post.js]: this name is confusing, it good to call it for example like ' screen '

File: homework-submission/photo-wall-app/src/components/post.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    this.setState({ form: 'add' })
+  }
+
+  switchToHome = () => {
+    this.setState({ form: 'list' })
+  }
+
+  render() {
+    console.log('postsState: ', this.state);
+    let lists = this.state.posts.map((post) => { return <PostsList key={post.id} posts={post} /> })
```
Comment (jalal75) [inline homework-submission/photo-wall-app/src/components/post.js]: can you find out why you get this warning, when you are add new photo:
Warning: Each child in an array or iterator should have a unique "key" prop.

File: homework-submission/photo-wall-app/src/components/add-from.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    })
+      .then(
+        this.setState({
+          photoUrl: '',
+          description: '',
+          form: 'list'
+        })
+      )
+    this.props.onAdd(this.state)
+    console.log('addFormProps ', this.props);
```
Comment (jalal75) [inline homework-submission/photo-wall-app/src/components/add-from.js]: Recheck this function..........

File: homework-submission/photo-wall-app/src/components/add-from.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
+import React, { Component } from 'react'
+import Service from './service'
```
Comment (jalal75) [inline homework-submission/photo-wall-app/src/components/add-from.js]: import Service from './Service'

File: homework-submission/photo-wall-app/src/components/post.js [hunk 7]
```diff
@@ -0,0 +1,63 @@
+import React, { Component } from 'react'
+import PostsList from './posts-list'
+import AddForm from './add-from'
+import Service from './service'
```
Comment (jalal75) [inline homework-submission/photo-wall-app/src/components/post.js]: import Service from './service'

**Comment by joudi6** [comment]
thank you, Jalal for your feedback I updated the homework regarding your advice.

**Comment by joudi6** [comment]
still working on Bonus assignment

**Pull request closed without merging**